### ISSUE TITLE
simple execution of arbitrary string rather than generated Operation 

### ISSUE THREAD

**Issue by nort3x** [apollographql/apollo-kotlin#4169]
**Use case**
manual and unittest-related tests
testing interceptors and engine without really having type-safe query-mutation-subscription in mind 

**Describe the solution you'd like**
for parsing a Map or even string itself suffice

**Comment by martinbonnin** [comment]
Hi @nort3x 👋 Would [HTTP level testing](https://www.apollographql.com/docs/kotlin/testing/mocking-http-responses/) work? 

```kotlin
// Enqueue HTTP responses
mockServer.enqueue("""{"data": {"random": 42}}""")

// Execute queries
assertEquals(
  42, 
  apolloClient.query(GetRandomQuery()).execute().data?.random
)
```

**Comment by nort3x** [comment]
hi @martinbonnin thanks for your reply,
interesting! i didn't know about mock-server but let me rephrase the issue description

suppose we have a setup for our client consist of different interceptors and even a custom http-engine
by default apollo client doesn't let us execute arbitrary input and it demands for existing *.graphql files to generate respective Query, Mutation and Subscription, by arbitrary i mean what you enter in a graphql live console (say graphiql for instance)

use cases?   unit-test and behavioral checks,
here is an example from [spring-graphql](https://github.com/spring-projects/spring-graphql/blob/main/samples/webmvc-http-security/src/test/java/io/spring/sample/graphql/WebMvcHttpSecuritySampleTests.java)
if we inspect it client is able to execute queries and inspect them using a path mechanism (this is an overkill!)
after all we are POSTing to the server why not be able to do it dynamically  without generated types

**Comment by martinbonnin** [comment]
Ah, gotcha! It's not just about tests, you want to execute runtime-defined operation without generated models, right? 

If that's the case, you'd have less troubles using an HTTP client like [`OkHttp`](https://square.github.io/okhttp/) directly. A big part of the value proposition of Apollo Kotlin is the generated models and type safety so if you don't need them, I'd say there's no need to use Apollo Kotlin? Or are you using something else (cache maybe?)?

**Comment by nort3x** [comment]
thank you, as i mentioned above use cases are for testing interceptors and HttpEngine beneath apollo client plus why such a soft cap is put on the chain?

**Comment by martinbonnin** [comment]
Hi 👋 Not sure I'm following. For `HttpEngine`, it doesn't know at all about GraphQL so you can test it without generated models. For interceptors, do you mind sharing an API you'd like to see? Maybe we can add it?

**Comment by martinbonnin** [comment]
@nort3x anything help we can help with here?

**Comment by nort3x** [comment]
I'm still not satisfied with current solution(which is revolving around testing HttpEngine separately from apollo client),
but i guess it's due personal preferences
so we can safely close this issue
thank you for your friendly support
